src/main/java/seedu/sellsavvy/logic/commands/ordercommands/DeleteOrderCommand.java

```diff
@@ -0,0 +1,77 @@
+package seedu.sellsavvy.logic.commands.ordercommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.sellsavvy.commons.util.CollectionUtil.requireAllNonNull;
+
+import javafx.collections.ObservableList;
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.commons.util.ToStringBuilder;
+import seedu.sellsavvy.logic.Messages;
+import seedu.sellsavvy.logic.commands.Command;
+import seedu.sellsavvy.logic.commands.CommandResult;
+import seedu.sellsavvy.logic.commands.exceptions.CommandException;
+import seedu.sellsavvy.model.Model;
+import seedu.sellsavvy.model.order.Order;
+import seedu.sellsavvy.model.order.OrderList;
+
+/**
+ * Deletes an order under a specified person.
+ */
+public class DeleteOrderCommand extends Command {
+
+    public static final String COMMAND_WORD = "deleteOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Deletes an order under the specified person. "
+            + "Parameters: INDEX (must be a positive integer) "
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_ORDER_SUCCESS = "Deleted Order: %1$s";
+
+    private final Index index;
+
+    /**
+     * Creates an DeleteOrderCommand to delete the specified {@param index of the order to be deleted}
+     */
+    public DeleteOrderCommand(Index index) {
+        requireAllNonNull(index);
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        ObservableList<Order> unmodifiableOrderList =
+                model.getSelectedPerson().get().getOrderUnmodifiableObservableList();
+        OrderList orderList = model.getSelectedPerson().get().getOrderList();
+
+        if (index.getZeroBased() >= orderList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ORDER_DISPLAYED_INDEX);
+        }
+
+        Order orderToDelete = unmodifiableOrderList.get(this.index.getZeroBased());
+        orderList.remove(this.index);
+        return new CommandResult(String.format(MESSAGE_DELETE_ORDER_SUCCESS, Messages.format(orderToDelete)));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof DeleteOrderCommand)) {
+            return false;
+        }
+
+        DeleteOrderCommand otherDeleteOrderCommand = (DeleteOrderCommand) other;
+        return index.equals(otherDeleteOrderCommand.index);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("index", index)
```

Comment:
based on your test case this should be target index

Reply:
Test case has been updated to reflect "index" string.